flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/runtime/batch/sql/join/ScalarQueryITCase.scala

```diff
@@ -46,6 +49,18 @@ class ScalarQueryITCase extends BatchTestBase {
     row(6, null)
   )
 
-}
+  @Before
+  override def before(): Unit = {
+    super.before()
+    registerCollection("l", l, INT_DOUBLE, "a, b")
+    registerCollection("r", r, INT_DOUBLE, "c, d")
+  }
 
+  @Test
+  def testScalarQuery(): Unit = {
```

Comment:
Also add a validation test for a multiple row agg?

Reply:
What is multiple row agg?
You mean use SingleValueAgg directly in sql? If that is what you mean, I prefer not add. Because SingleValueAgg is just add for scalar sub query, we don't recommend user to use in sql.

Reply:
No. I mean something like `SELECT * FROM l WHERE a = (SELECT c FROM r WHERE c > 3)`, so that the inner select produces more than 1 record. We should expect an validation failure on this case.

Reply:
Got it, I will rebase and add it.